graders/python/python_autograder/pl_execute.py

```diff
@@ -76,7 +76,7 @@ def execute_code(
 
     # Read student code (and transform if necessary) and append leading/trailing code
     with open(fname_student, encoding="utf-8") as f:
-        filename, extension = path.splitext(fname_student)
+        _filename, extension = path.splitext(fname_student)
```

Comment:
I think we can avoid declaring a variable at all here (please test):

```suggestion
        extension = path.splitext(fname_student)[1]
```

Ditto elsewhere if you like this.

Reply:
Generally, I prefer tuple unpacking (though not a super strong preference)

Reply:
IMO tuple unpacking is great if you're unpacking multiple values. I don't see the benefit when there's just a single value that will be used.

Reply:
Just to use the language feature because I like it 😂

Reply:
I actually prefer tuple unpacking here as it provides additional information as to how many return values there are.